src/Components/ScrivitoExtensions/AdvancedEnumEditor.tsx

```diff
@@ -0,0 +1,58 @@
+import { connect, uiContext } from 'scrivito'
+import { ensureString } from '../../utils/ensureString'
+import './AdvancedEnumEditor/AdvancedEnumEditor.scss'
+
+type EnumOption = {
+  value: string
+  title: string
+  description?: string
+  icon?: string
+}
+
+type AdvancedEnumEditorProps = {
+  attributeValue: unknown
+  options: EnumOption[]
+  readOnly: boolean
+  updateAttributeValue: (value: string) => void
+}
+
+export const AdvancedEnumEditor = connect(function AdvancedEnumEditor({
+  attributeValue,
+  options,
+  readOnly,
+  updateAttributeValue,
+}: AdvancedEnumEditorProps) {
+  const theme = (uiContext() || { theme: null }).theme
+  if (!theme) return null
+
+  const currentValue = ensureString(attributeValue) || options[0]?.value
+
+  return (
+    <div className={`advanced-enum-editor scrivito_${theme}`}>
+      <div className="enum_attribute">
+        {options.map((option) => (
+          <button
+            key={option.value}
+            aria-current={currentValue === option.value}
+            className={
+              currentValue === option.value
+                ? 'enum_attribute_active'
+                : undefined
+            }
+            disabled={readOnly}
+            onClick={() => updateAttributeValue(option.value)}
+            title={option.description}
```

Comment:
* [ ] FYI Naming this option "description" is misleading, as a description should be easily discoverable. Maybe "tooltip"?